src/driver/src/DefaultDriver.cpp

```diff
@@ -188,58 +197,82 @@ std::string normalizeOutputFile(const std::string& str) {
 }  // namespace
 
 
-genny::driver::DefaultDriver::ProgramOptions::ProgramOptions(int argc, char** argv) {
+DefaultDriver::ProgramOptions::ProgramOptions(int argc, char** argv) {
     namespace po = boost::program_options;
 
-    po::options_description description{u8"🧞‍ Allowed Options 🧞‍"};
+    po::options_description progDescription{u8"🧞‍ Allowed Options 🧞‍"};
     po::positional_options_description positional;
 
     // clang-format off
-    description.add_options()
-        ("help,h",
-            "Show help message")
-        ("list-actors",
-            "List all actors available for use")
-        ("dry-run",
-            "Exit before the run step---"
-            "this may still make network connections during workload initialization")
-        ("metrics-format,m",
+    const auto runModeHelp = u8R"(
+Genny run modes
+    normal
+        Run the workload normally; default mode if no run mode is specified
+
+    dry-run
+        Exit before the run step -- this may still make network connections during workload initialization
+
+    evaluate
+        Print the evaluated YAML workload file with minimal validation
+
+    list-actors
+        List all actors available for use
+    )";
+
+    progDescription.add_options()
+            ("help,h",
+             "Show help message")
+            ("run-mode", po::value<std::string>()->default_value("normal"),
+             runModeHelp)
```

Comment:
do we want to require the `--run-mode` param or can it just be the first positional param similar to `git` et al. E.g.

```sh
genny list-actors
genny run # aka "normal"
```

This would change the "run mode" to be more of a "command" or "action". It seems clunky to require an option flag to tell it what to do. Imagine if you had to type `git --action checkout` 😸
I don't have a strong opinion here so please push back if you think it's not worth it.

Reply:
Funny you should ask! Everything before the "[don't use subcommands](https://github.com/mongodb/genny/pull/196/commits/205956bdf52ddc2113ad9790f93341d1ab8dffda)" commit used subcommands as you described.

The problem is the workload yaml is also a positional argument and genny can't distinguish between the two easily. I didn't want to want to have genny guess whether the first argument is a workload file or a subcommand.

What do you think about removing the workload positional argument?

Reply:
> genny can't distinguish between the two easily

it seems like it should always assume that the first argument is the subcommand and any after that are positional arguments to the subcommand. So running just `genny foo.yaml` shouldn't work because `foo.yaml` isn't a subcommand but `genny run foo.yaml` should assume subcommand is run and the first positional arg is foo.yaml.

But if you're saying that boost program options doesn't let you easily express this then it does seem reasonable to only allow one positional option and have that be the subcommand and to give all other parameters via options e.g. `genny run --workload-file foo.yaml`. I've seen some tools do this so it's not totally unheard-of, but tools like git and aws-cli handle multiple positional arguments and shift them with ease.

(ultimately users interact with the genny binary so infrequently this is edging closer to bikeshedding so please just do what you feel is right - biggest thing for me is having one distinct runmode option that all the various list-actors/normal/evaluate/* all fall under.  Whether it's a --option or a positional argument and whether subsequent things are positional or not doesn't end up affecting us too much imho.